### PR TITLE
Cleanup StringInfo tests

### LINKED ISSUE #2336: Rename tests in System.Globalization
About 40-50% of the tests in System.Globalization do not have self descriptive names. They are currently named PosTest, NegTest etc.

### PR DESCRIPTION
**Pull request by hughbe**
This enabled me to combine multiple files that test the same method into
single files, without creating additional complexity.

### CODE REVIEW

File: src/System.Globalization/tests/StringInfo/StringInfoCtor.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+using System.Globalization;
+using Xunit;
+
+namespace System.Globalization.Tests
+{
+    public class StringInfoCtorTests
+    {
+        private const int MinStringLength = 8;
+        private const int MaxStringLength = 256;
+        private static readonly RandomDataGenerator s_RandomDataGenerator = new RandomDataGenerator();
```
Comment (stephentoub) [inline src/System.Globalization/tests/StringInfo/StringInfoCtor.cs]: Nit: s_RandomDataGenerator => s_randomDataGenerator

File: src/System.Globalization/tests/StringInfo/StringInfoGetNextTextElement.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+using System.Globalization;
+using Xunit;
+
+namespace System.Globalization.Tests
+{
+    public class StringInfoGetNextTextElement
+    {
+        private const int MinStringLength = 8;
+        private const int MaxStringLength = 256;
+        private static readonly RandomDataGenerator s_RandomDataGenerator = new RandomDataGenerator();
```
Comment (stephentoub) [inline src/System.Globalization/tests/StringInfo/StringInfoGetNextTextElement.cs]: Same nit (whereever else this shows up, too).

File: src/System.Globalization/tests/StringInfo/StringInfoGetNextTextElement.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        private static readonly Random s_Random = new Random(-55);
```
Comment (stephentoub) [inline src/System.Globalization/tests/StringInfo/StringInfoGetNextTextElement.cs]: Ditto

**Comment by stephentoub** [comment]
Just a few naming nits, and since the code is now better than it was and our CI system is currently down and this had a successful CI run before then, I'm going to go ahead and merge it.

**Comment by hughbe** [comment]
@stephentoub Glad this was merged, thanks. Do you suggest I bother sending in a PR to fix these nits

**Comment by stephentoub** [comment]
> Do you suggest I bother sending in a PR to fix these nits

Probably not worth it at this point.  If more changes get made to these files, then sure.

**Comment by hughbe** [comment]
Understood.

**Pull request merged**